web/markdown/rules.react.js

```diff
@@ -47,8 +71,105 @@ function linkRules(): MarkdownRuleSpec {
+    link: {
+      ...linkMarkdownRules.simpleMarkdownRules.link,
+      match: SimpleMarkdown.defaultRules.link.match,
+    },
+    blockQuote: {
+      ...SimpleMarkdown.defaultRules.blockQuote,
+      // match end of blockQuote by either \n\n or end of string
+      match: SimpleMarkdown.blockRegex(/^( *>[^\n]+(\n[^\n]+)*\n*)+(\n{2,}|$)/),
+      parse(
+        capture: SimpleMarkdown.Capture,
+        parse: SimpleMarkdown.Parser,
+        state: SimpleMarkdown.State,
+      ) {
+        var content = capture[0].replace(/^ *> ?/gm, '');
+        return {
+          // remove new line from captured string
+          content: parse(content.trim(), state),
+        };
+      },
+      // eslint-disable-next-line react/display-name
+      react: (
+        node: SimpleMarkdown.SingleASTNode,
+        output: SimpleMarkdown.Output<SimpleMarkdown.ReactElement>,
+        state: SimpleMarkdown.State,
+      ) => (
+        <blockquote key={state.key} className={css.block}>
+          {output(node.content, state)}
+        </blockquote>
+      ),
+    },
+    inlineCode: {
+      ...SimpleMarkdown.defaultRules.inlineCode,
+      // eslint-disable-next-line react/display-name
+      react: (
+        node: SimpleMarkdown.SingleASTNode,
+        output: SimpleMarkdown.Output<SimpleMarkdown.ReactElement>,
+        state: SimpleMarkdown.State,
+      ) => (
+        <code key={state.key} className={css.inlineCode}>
+          {node.content}
+        </code>
+      ),
+    },
+    em: SimpleMarkdown.defaultRules.em,
+    strong: SimpleMarkdown.defaultRules.strong,
+    del: SimpleMarkdown.defaultRules.del,
+    underline: {
+      order: SimpleMarkdown.defaultRules.em.order - 0.5,
+      match: SimpleMarkdown.inlineRegex(/^__([\s\S]+?)__(?!_)/),
+      parse: (
+        capture: SimpleMarkdown.Capture,
+        parse: SimpleMarkdown.Parser,
+        state: SimpleMarkdown.state,
+      ) => ({
+        content: parse(capture[1], state),
+      }),
+      // eslint-disable-next-line react/display-name
+      react: (
+        node: SimpleMarkdown.SingleASTNode,
+        output: SimpleMarkdown.Output<SimpleMarkdown.ReactElement>,
+        state: SimpleMarkdown.State,
+      ) => <u key={state.key}>{output(node.content, state)}</u>,
+    },
+    // heading: {
+    //   ...SimpleMarkdown.defaultRules.heading,
+    //   match: SimpleMarkdown.inlineRegex(/^ *(#{1,6} )([^\n]+?) *(?:\n+|$)/),
+    // },
+    mailto: SimpleMarkdown.defaultRules.mailto,
+    codeBlock: {
+      ...SimpleMarkdown.defaultRules.codeBlock,
+      match: SimpleMarkdown.blockRegex(/^(?: {4}[^\n]+\n*)+(?:\n *)/),
+      parse(capture: SimpleMarkdown.Capture) {
+        var content = capture[0].replace(/^ {4}/gm, '');
+        return {
+          lang: undefined,
+          content: content,
+        };
+      },
+    },
+    fence: {
+      ...SimpleMarkdown.defaultRules.fence,
+      match: SimpleMarkdown.blockRegex(/`{3}(?:(.*$)\n)?([\s\S]*)`{3}/m),
```

Comment:
Nice job with this RegEx! You significantly simplified the code. Some changes I think might make sense:

1. Do you need the `m` modifier at the end? I think it only modifies the behavior of beginning (`^`) and end (`$`) characters, which you aren't using.
2. Can we also match the tilde character? We can use `\1` to make sure we don't match ~~~```:

```js
/(`{3,}|~{3,})(?:(.*$)\n)?([\s\S]*)\1/
```

Since we're now adding a capture group you'll need to update the `parse` function

Reply:
I noticed that all of `simple-markdown` RegExes start with `^`. I'm working if matching in the middle of the text will break `simple-markdown`. I think [this line](https://github.com/Khan/simple-markdown/blob/master/simple-markdown.js#L464) assumes that the capture happens at the start of the text